Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -313,20 +313,41 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
       return information;
     }
 
-    private string FormatResourceCount(int nodeResourceCount) {
+    private static readonly int SignificativeDigits = 3;
+
+    public static long RoundToSignificativeDigits(long nodeResourceCount) {
+      var nDigits = ("" + nodeResourceCount).Length;
+      if (nDigits > SignificativeDigits) {
+        var toRemove = (long)Math.Pow(10, nDigits - SignificativeDigits);
+        nodeResourceCount += toRemove / 2;
+        nodeResourceCount -= (nodeResourceCount % toRemove);
+      }
+
+      return nodeResourceCount;
+    }
+
+    public static string FormatResourceCount(long nodeResourceCount) {
```

Comment:
This is so much complicated code, even though .NET has quite a lot of number formatting utilities. Wouldn't `nodeResourceCount.ToString("G4")` give you more or less the same result? https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings

Reply:
It would but without the prefixed K, M, G which are very handy.

Reply:
Is that easier to read than `e+XXX` notation? 

It might be easier to compare `8.72e+005` with `8.72e+006` than it is to compare `872K` with `8.72M`

Reply:
That would be much simpler to encode, true. But is it more readable?

For me the scientific notation is like.... 5.25e005 RU? What is that? Oh that's just 525K RU? Should have told earlier :-) It's even worse when we switch between the scientific notation to the normal notation, it makes it hard to compare. For example 525*10 = 5.25e003. Hard to see where the 003 comes from or compare the two right away.
Or if we delayed the scientific notation, 525, 5250, 52500, 5.25e005.
On the other side, when the unit is the same, (which happens most of the times),  the number of digits before the comma immediately tells about the magnitude, it's a visual comparison, not something to be interpreter from numbers like 5, 6.

Here are some comparisons:
* On the Windows file system as of today, everything is expressed in KB, even if it's millions.
* Total valuations of businesses are always expressed in terms of K (kilos) M (millions) or B (billions)

Happy to discuss this further if you think the current unit system can be enhanced.

Reply:
> It's even worse when we switch between the scientific notation to the normal notation, it makes it hard to compare.

I agree, I would always do the scientific one.

> For me the scientific notation is like.... 5.25e005 RU? What is that? Oh that's just 525K RU? Should have told earlier :-)

I don't think anyone using resource counts in Dafny will have trouble reading scientific notation.

I you want to compare magnitude, comparing e00X with e00Y, versus comparing two instances of K/M/G/etc. + a number for everything between the 1000s, than the former seems simpler to me.